r/R/io.R

```diff
@@ -241,7 +241,7 @@ mmap_open <- function(path, mode = c("read", "write", "readwrite")) {
 make_readable_file <- function(file, mmap = TRUE, compression = NULL, filesystem = NULL) {
   if (inherits(file, "SubTreeFileSystem")) {
     filesystem <- file$base_fs
-    file <- file$base_path
+    file <- gsub("/$", "", file$base_path)
```

Comment:
nit: no need for global substitution

```suggestion
    file <- sub("/$", "", file$base_path)
```

Reply:
This is also worth leaving a code comment to explain why